src/extremeweatherbench/data_loader.py

```diff
@@ -0,0 +1,169 @@
+import xarray as xr
+from extremeweatherbench import config, utils
+import pandas as pd
+import logging
+import dataclasses
+from typing import Tuple, Optional
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+def open_forecast_dataset(
+    eval_config: config.Config,
+    forecast_schema_config: config.ForecastSchemaConfig,
+) -> xr.Dataset:
+    """Open the forecast dataset specified for evaluation.
+
+    If a URI is provided (e.g. s3://bucket/path/to/forecast), the filesystem
+    will be inferred from the provided source (in this case, s3). Otherwise,
+    the filesystem will assumed to be local.
+    Args:
+        eval_config: The evaluation configuration.
+        forecast_schema_config: The forecast schema configuration.
+
+    Returns:
+        The opened forecast dataset.
+    """
+    logger.debug("Opening forecast dataset")
+    if "zarr" in eval_config.forecast_dir:
+        forecast_ds = xr.open_zarr(eval_config.forecast_dir, chunks="auto")
+    elif (
+        "parq" in eval_config.forecast_dir
+        or "json" in eval_config.forecast_dir
+        or "parquet" in eval_config.forecast_dir
+    ):
+        forecast_ds = open_kerchunk_reference(eval_config)
+    else:
+        raise TypeError(
+            "Unknown file type found in forecast path, only json, parquet, and zarr are supported."
+        )
+    forecast_ds = _rename_forecast_dataset(forecast_ds, forecast_schema_config)
+    forecast_ds = utils.maybe_convert_dataset_lead_time_to_int(forecast_ds)
+
+    return forecast_ds
+
+
+def open_kerchunk_reference(
+    eval_config: config.Config,
+) -> xr.Dataset:
+    """Open a dataset from a kerchunked reference file in parquet or json format.
+    This has been built for the CIRA MLWP S3 bucket's data (https://registry.opendata.aws/aiwp/),
+    but can work with other data in the future. Currently only supports CIRA data unless
+    schema is identical to the CIRA schema.
+
```

Comment:
Keep this function as generic as possible. You can write a helper function, `open_cira_archive`, which _uses_ this function by creating and appropriate `eval_config`. But we should not be hard-coding anything about specific datasets - instead, force ourselves to use the appropriate convention and interface that other users have to.